lab-raziyeh/test/book-test.js

```diff
@@ -0,0 +1,169 @@
+            .end((err, res) => {
+              if(err) return done(err);
+              expect(res.status).to.equal(200);
+              expect(res.body.id).to.equal(tempBook.id);
+              expect(res.body.title).to.equal(tempBook.title);
+              expect(res.body.page).to.equal(tempBook.page);
+              done();
+            });
+      });
+    });
+
+    describe('testing GET /api/book invalid id  in the request', function(){
+      it('expecting to return an error with unValid Id - GET', function(done){
+        request.get(`${url}/api/book?id=1234`)
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          done();
+        });
+      });
+    });
+
+    describe('testing GET /api/book no id provided in the request', function() {
+      it('expecting to return an error with no id - GET', function(done){
+        request.get(`${url}/api/book`)
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          expect(res.text).equal('BadRequestError');
+          expect(res.err).not.be.null;
+          done();
+        });
+      });
+    });
+  });
+
+  describe('POST requests tests', function() {
+    describe('testing POST with a valid body /api/book', function() {
+
+      after(done => {
+        if(tempBook) {
+          Book.deleteItem('book', tempBook.id)
+          .then(() => done())
+          .catch(err => done(err));
+        }
+      });
+
+      it('expecting save a new Book - POST', (done) => {
+        request.post(`${url}/api/book`)
+        .send(tempBook)
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(!!res.body.id).to.equal(true);
+          expect(res.body.title).to.equal(tempBook.title);
+          expect(res.body.author).to.equal(tempBook.author);
+          this.tempBook = res.body;
+          done();
+        });
+      });
+    });
+
+    describe('testing POST /api/book', function(){
+      it('expecting respond with bad request - POST', function(done){
+        request.post(`${url}/api/book`)
+        .send({ISBN:'445555444'})
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          expect(res.text).to.be.equal('BadRequestError');
+          done();
+        });
+      });
+    });
+  });
+
+  describe('Testing PUT requests to api/book', function(){
+  // PUT - test 200, response body like {<data>} for a post request with a valid body
+    describe('valid id and body', function() {
+      before( done => {
+        Book.createItem(exampleBook)
+        .then(book => {
+          this.tempBook = book;
+          done();
+        })
+        .catch(err => done(err));
+      });
+
+      after(done => {
+        if(this.tempBook) {
+          Book.deleteItem(this.tempBook.id)
+         .then(() => done())
+         .catch(err => done(err));
+        }
+      });
+
+      // it('should return an book update', (done) => {
```

Comment:
@Raziyehbazargan this should be removed from your `[master]` branch - large, uncommented blocks, should be contained on a separate branch - ie: `features/testing_stuff`